src/core/utils/useWorkUrl.ts

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from "react";
+import { useComplexSearchWithPaginationQuery } from "../dbc-gateway/generated/graphql";
+import { useUrls } from "./url";
+import { constructMaterialUrl } from "./helpers/url";
+import { Pid, WorkId } from "./types/ids";
+import { getMaterialTypes } from "./helpers/general";
+import { Manifestation } from "./types/entities";
+
+type UseWorkUrlParams = {
+  identifier?: string | null; // ISBN for digital materials
+  pid?: Pid | null; // PID for physical materials
+  materialType?: string | null;
+};
+
+type UseWorkUrlResult = {
+  workUrl: URL | null;
+  isLoading: boolean;
+};
+
+const useWorkUrl = ({
```

Comment:
I don't know if it's a react thing or not, but I feel like these files would be much easier to understand if they had a `docblock` that explains in rough terms what the method does.

I know that you're just following the style of the rest of the project, but seeing as this is a new file, it seems an ideal place to add it